### PR TITLE
Do only build from main branch

### LINKED ISSUE #608: Website is built from any branch
As it seems with the program for MEC2024, the website gets build from any branch, not only from changes on main. This should be changed.

### PR DESCRIPTION
**Pull request by musicEnfanthen**
This PR adds a branch filter to the build script, so the website gets only build when there are pushes to the main branch.

Fixes #608

### CODE REVIEW

**Review by rettinghaus — APPROVED** [review]
Please do not remove the newline at the end. 

> [!Warning]
> The POSIX standard requires the last line to end with a new line character

**Comment by musicEnfanthen** [comment]
That's automatic, did not do anything to that line ending. Will try to undo.

File: .github/workflows/jekyll.yml [hunk 1]
```diff
@@ -23,4 +24,4 @@ jobs:
       - uses: helaili/jekyll-action@c1527523361ec3ecc54b2371ddef44826e28c0f5 # ratchet:helaili/jekyll-action@v2
         with:
           token: ${{ secrets.GITHUB_TOKEN }}
-          target_branch: 'gh-pages'
\ No newline at end of file
+          target_branch: 'gh-pages'
```
Comment (musicEnfanthen) [inline .github/workflows/jekyll.yml:27]:
```suggestion
          target_branch: 'gh-pages'
          
```

**Comment by rettinghaus** [comment]
Some hints: https://docs.github.com/en/get-started/getting-started-with-git/configuring-git-to-handle-line-endings

**Comment by musicEnfanthen** [comment]
Thanks. The thing is, I've created this PR purely on GitHub UI. No local git involved. ❓

**Comment by musicEnfanthen** [comment]
Superseded by #610

**Comment by musicEnfanthen** [comment]
PS: It seems that the default line ending used by GH is CRLF, see https://github.com/actions/runner/issues/1462#issuecomment-1030124116:

> Hi @marcellp, this is how pull requests work on GitHub, it is not specific to the runner. The GitHub API uses \r\n as line breaks within pull requests, which you can see with the /repos/{org}/{repo}/pulls/{pr_number} API; the body in the Actions event matches the body of the PR, as it is stored. GitHub Actions does not mutate this, it gives it to you verbatim. If you want a modified version of the PR body, you'll need to do it yourself.

**Pull request closed without merging**